lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
Yes, it will. In that case, we just won't get the optimization. 

---
In reply to: [78790053](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78790053) [](ancestors = 78790053)